sdk/core/core-http/src/serviceClient.ts

```diff
@@ -364,15 +364,16 @@ export class ServiceClient {
                     queryParameterValue[index] = item == undefined ? "" : item.toString();
                   }
                 }
-              } else {
-                queryParameterValue = queryParameterValue.join(queryParameter.collectionFormat);
               }
             }
             if (!queryParameter.skipEncoding) {
               if (Array.isArray(queryParameterValue)) {
                 for (const index in queryParameterValue) {
                   queryParameterValue[index] = encodeURIComponent(queryParameterValue[index]);
                 }
+                if (queryParameter.collectionFormat !== QueryCollectionFormat.Multi) {
+                  queryParameterValue = queryParameterValue.join(queryParameter.collectionFormat);
+                }
```

Comment:
Don't we need to do the above when `queryParameter.skipEncoding` is `true` as well?

Reply:
Yes. Modified the code and added a test case for the same